x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -482,6 +495,93 @@ func (k Keeper) TokenLogicModuleGlobalMint(
+
+	// Do not process the reimbursement request if there is no global inflation.
+	if MintPerClaimedTokenGlobalInflation == 0 {
+		logger.Warn("global inflation is set to zero. Skipping Global Mint Reimbursement Request TLM.")
+		return nil
+	}
+
+	// Determine how much new uPOKT to mint based on global inflation
+	newMintCoin, _ := calculateGlobalPerClaimMintInflationFromSettlementAmount(actualSettlementCoin)
+	if newMintCoin.Amount.Int64() == 0 {
+		return tokenomicstypes.ErrTokenomicsMintAmountZero
+	}
+
+	// Update the application's on-chain stake
+	// This should not fall below zero since `ensureClaimAmountLimits` should have
+	// already checked and adjusted the settlement amount so that the application
+	// stake covers the global inflation.
+	newAppStake := application.Stake.Sub(newMintCoin)
+	application.Stake = &newAppStake
+	logger.Info(fmt.Sprintf("updated application %q stake to %s", application.Address, newAppStake))
+
+	// Send the global per claim mint inflation uPOKT from the application module
+	// account to the tokenomics module account.
+	if err := k.bankKeeper.SendCoinsFromModuleToModule(
+		ctx, apptypes.ModuleName, tokenomicstypes.ModuleName, sdk.NewCoins(newMintCoin),
+	); err != nil {
+		return tokenomicstypes.ErrTokenomicsApplicationReimbursementRequestFailed.Wrapf(
+			"sending %s from the application module account to the tokenomics module account: %v",
+			newMintCoin, err,
+		)
+	}
+	logger.Info(fmt.Sprintf(
+		"sent (%s) from the application module account to the tokenomics module account",
+		newMintCoin,
+	))
+
+	// Send the global per claim mint inflation uPOKT from the tokenomics module
+	// account to PNF/DAO.
+	daoAccountAddr, err := cosmostypes.AccAddressFromBech32(k.GetAuthority())
+	if err != nil {
+		return tokenomicstypes.ErrTokenomicsApplicationReimbursementRequestFailed.Wrapf(
+			"getting PNF/DAO address: %v",
+			err,
+		)
+	}
+	if err := k.bankKeeper.SendCoinsFromModuleToAccount(
+		ctx, tokenomicstypes.ModuleName, daoAccountAddr, sdk.NewCoins(newMintCoin),
+	); err != nil {
+		return tokenomicstypes.ErrTokenomicsApplicationReimbursementRequestFailed.Wrapf(
+			"sending %s from the tokenomics module account to the PNF/DAO account: %v",
+			newMintCoin, err,
+		)
+	}
+
+	// Prepare and emit the event for the application being overcharged.
+	reimbursementRequestEvent := &tokenomicstypes.EventApplicationReimbursementRequest{
+		ApplicationAddr:      application.Address,
+		SupplierOperatorAddr: supplier.OperatorAddress,
+		SupplierOwnerAddr:    supplier.OwnerAddress,
+		ServiceId:            service.Id,
+		SessionId:            sessionHeader.SessionId,
+		Amount:               &newMintCoin,
+	}
+
+	eventManger := cosmostypes.UnwrapSDKContext(ctx).EventManager()
+	if err := eventManger.EmitTypedEvent(reimbursementRequestEvent); err != nil {
+		return tokenomicstypes.ErrTokenomicsEmittingEventFailed.Wrapf(
+			"(%+v): %s",
+			reimbursementRequestEvent, err,
+		)
```

Comment:
I've been following this pattern but am open to feedback:
```suggestion
		err = tokenomicstypes.ErrTokenomicsEmittingEventFailed.Wrapf(
			"(%+v): %s",
			reimbursementRequestEvent, err,
		)
		logger.Error(err.Error())
		return err
```